merkledag/merkledag.go

```diff
@@ -146,21 +145,58 @@ func FindLinks(links []key.Key, k key.Key, start int) []int {
 	return out
 }
 
+func (ds *dagService) GetMany(ctx context.Context, keys []key.Key) (<-chan *Node, <-chan error) {
+	out := make(chan *Node)
+	errs := make(chan error, 1)
+	blocks := ds.Blocks.GetBlocks(ctx, keys)
+	var count int
+
+	go func() {
+		defer close(out)
+		defer close(errs)
+		for {
+			select {
+			case b, ok := <-blocks:
+				if !ok {
+					if count != len(keys) {
+						errs <- fmt.Errorf("failed to fetch all nodes")
+					}
+					return
+				}
+				nd, err := Decoded(b.Data)
+				if err != nil {
+					errs <- err
+					return
+				}
+				select {
+				case out <- nd:
```

Comment:
hrm, i guess the improved simplicity there would outweigh any performance gains this might have. I generally avoid buffering channels, but it makes sense here